eventstreamrpc/include/aws/eventstreamrpc/EventStreamClient.h

```diff
@@ -0,0 +1,605 @@
+        {
+          public:
+            /**
+             * This callback is only invoked upon receiving a CONNECT_ACK with the
+             * CONNECTION_ACCEPTED flag set by the server. Therefore, once this callback
+             * is invoked, the `ClientConnection` is ready to be used for sending messages.
+             */
+            virtual void OnConnectCallback();
+            /**
+             * Invoked upon connection shutdown. `errorCode` will specify
+             * shutdown reason. A graceful connection close will set `errorCode` to
+             * `AWS_ERROR_SUCCESS` or 0.
+             */
+            virtual void OnDisconnectCallback(int errorCode);
+            /**
+             * Invoked upon receiving an error. Use the return value to determine
+             * whether or not to force the connection to close. Keep in mind that once
+             * closed, the `ClientConnection` can no longer send messages.
+             */
+            virtual bool OnErrorCallback(int errorCode);
+            /**
+             * Invoked upon receiving a ping from the server. The `headers` and `payload`
+             * refer to what is contained in the ping message.
+             */
+            virtual void OnPingCallback(
+                const Crt::List<EventStreamHeader> &headers,
+                const Crt::Optional<Crt::ByteBuf> &payload);
+        };
+
+        class AWS_EVENTSTREAMRPC_API ClientContinuationHandler
+        {
+          public:
+            /**
+             * Invoked when a message is received on this continuation.
+             */
+            virtual void OnContinuationMessage(
+                const Crt::List<EventStreamHeader> &headers,
+                const Crt::Optional<Crt::ByteBuf> &payload,
+                MessageType messageType,
+                uint32_t messageFlags) = 0;
+            /**
+             * Invoked when the continuation is closed.
+             *
+             * Once the continuation is closed, no more messages may be sent or received.
+             * The continuation is closed when a message is sent or received with
+             * the TERMINATE_STREAM flag, or when the connection shuts down.
+             */
+            virtual void OnContinuationClosed() = 0;
+        };
+
+        enum EventStreamRpcError
+        {
+            /* If error messages are added to `aws_event_stream_errors`, this will need to be updated. */
+            EVENT_STREAM_RPC_SUCCESS = 0,
+            EVENT_STREAM_RPC_NULL_PARAMETER,
+            EVENT_STREAM_RPC_INITIALIZATION_ERROR,
+            EVENT_STREAM_RPC_CONNECTION_CLOSED_BEFORE_CONNACK,
+            EVENT_STREAM_RPC_UNKNOWN_PROTOCOL_MESSAGE,
+            EVENT_STREAM_RPC_UNMAPPED_DATA,
+            EVENT_STREAM_RPC_UNSUPPORTED_CONTENT_TYPE,
+            EVENT_STREAM_RPC_STREAM_CLOSED_ERROR,
+            EVENT_STREAM_RPC_UNEXPECTED_ERROR,
+            EVENT_STREAM_RPC_CRT_ERROR
+        };
+
+        struct EventStreamRpcStatus
```

Comment:
This is too broad of a name and makes it unclear. EventStreamRpcResult is a suggestion